src/clj_cbor/core.clj

```diff
@@ -116,11 +107,11 @@
    (encode-seq default-codec values))
   ([encoder values]
    (let [buffer (ByteArrayOutputStream.)]
-     (with-open [output (data-output-stream buffer)]
+     (with-open [output (DataOutputStream. buffer)]
        (encode-seq encoder output values))
      (.toByteArray buffer)))
   ([encoder ^OutputStream output values]
-   (let [data-output (data-output-stream output)]
+   (let [data-output (DataOutputStream. output)]
```

Comment:
This would double-wrap an existing `DataOutputStream`, which is probably fine in practice but not ideal.